src/Psalm/Internal/Analyzer/FunctionLike/ReturnTypeAnalyzer.php

```diff
@@ -721,8 +721,22 @@ private static function addOrUpdateReturnType(
             $function_like_analyzer->getMethodId(),
             $function
         );
+
+        $is_final = true;
+        $fqcln = $source->getFQCLN();
+        if ($fqcln !== null) {
+            $class_storage = $source->getCodebase()->classlike_storage_provider->get($fqcln);
```

Comment:
You may want to have `$codebase = $source->getCodebase();` up top, then reference that variable in the places below.